packages/nimble-components/src/select/index.ts

```diff
@@ -67,27 +201,164 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        if (this.filterMode === FilterMode.none) {
+            return super.keydownHandler(e);
+        }
+
+        const key = e.key;
+        if (e.ctrlKey || e.shiftKey) {
+            return true;
+        }
+
+        switch (key) {
+            case keySpace: {
+                // when dropdown is open allow user to enter a space for filter text
+                // (calling super method will close dropdown)
+                if (!this.open) {
+                    super.keydownHandler(e);
+                }
+                break;
+            }
+
+            case keyEnter: {
+                this.updateSelectedIndexFromFilteredSet();
+                super.keydownHandler(e);
+                this.focus();
+                break;
+            }
+            case keyEscape: {
+                if (this.committedSelectedOption) {
+                    // clear filteredOptions as call to `super.keydownHandler` will process
+                    // "options" and not "_options"
+                    this.filteredOptions = [];
+                    this.clearSelection();
+                    this.selectedIndex = this._options.indexOf(
+                        this.committedSelectedOption
+                    );
+                }
+                super.keydownHandler(e);
+                this.focus();
+                break;
+            }
+
+            default: {
+                super.keydownHandler(e);
+            }
+        }
+        return true;
+    }
+
+    protected override openChanged(
+        prev: boolean | undefined,
+        next: boolean
     ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+        super.openChanged(prev, next);
+        if (!this.$fastController.isConnected) {
+            return;
+        }
+
+        if (this.open) {
+            this.filterOptions();
+            window.requestAnimationFrame(() => {
+                this.searchInput?.focus();
+            });
+        } else {
+            this.filter = '';
+            if (this.searchInput) {
+                this.searchInput.value = '';
+            }
+        }
+
+        if (this.open && this.committedSelectedOption) {
+            this.committedSelectedOption.selected = true;
+        }
+    }
+
+    /**
+     * Filter available options by text value.
+     *
+     * @public
+     */
+    private filterOptions(): void {
+        const filter = this.filter.toLowerCase();
+
+        if (filter) {
+            this.filteredOptions = this._options.filter(option => {
+                return diacriticInsensitiveStringNormalizer(
+                    option.text
+                ).includes(diacriticInsensitiveStringNormalizer(filter));
+            });
+        } else {
+            this.filteredOptions = this._options;
         }
+
+        this._options.forEach(o => {
+            o.hidden = !this.filteredOptions.includes(o);
+        });
```

Comment:
A client used to be able to hide an option in the select by marking it as `hidden`. I don't think that works anymore (or at least I couldn't get it to work by modifying the DOM in the storybook page published with this branch).

What are the expectations about this behavior?

Reply:
Hmmm...good question. The current behavior ends up matching what happens with the `Combobox`, which is to say if a client has marked an option as `hidden` and then proceeds to filter the options, the hidden option will be revealed (and stay revealed even if the filter is removed). It will be good to hear from others on whether this is acceptable.

Reply:
We discussed this off-line, and we've decided that we will follow the behavior of the `Combobox` but still document it.